<?xml version="1.0"?>
<launch>
  <!-- MoveIt options -->
  <arg name="pipeline" default="ompl" doc="Planning pipeline to use with MoveIt"/>

  <!-- Gazebo options -->
  <arg name="use_rviz" default="false" />
  <arg name="gazebo_gui" default="true" doc="Start Gazebo GUI"/>
  <arg name="paused" default="false" doc="Start Gazebo paused"/>
  <arg name="world_name" default="worlds/empty.world" doc="Gazebo world file"/>
  <!-- <arg name="world_pose" default="-x 0 -y 0 -z 0 -R 0 -P 0 -Y 0" doc="Pose to spawn the robot at"/> -->
    <!-- Gazebo options -->

    <!-- Start Gazebo paused to allow the controllers to pickup the initial pose -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch" pass_all_args="true">
    <arg name="paused" value="true"/>
  </include>
  <!-- <group ns="/lego_gazebo"> -->
    <include file="$(find hierarchial_simulation)/launch/lego_setup_BeaconTower.launch"/>
    <!-- <include file="$(find hierarchial_simulation)/launch/lego_setup.launch"/> -->
    <!-- <node pkg="hierarchial_simulation" name="lego_state_refresher" type="demo_human_fanuc_lego.py" output="screen">
      <param name="config_json" value="$(find hierarchial_simulation)/config/ICL.json"/>
    </node> -->
  <!-- </group> -->


  <!-- <group ns="/human_gazebo">
    <arg name = "human" default="human"/>
    <arg name="model" value="human_gazebo"/>

    <arg name="world_pose" value="-x 0.4 -y 0.4 -z 0.1 -R 0 -P 0 -Y 3.14159" doc="Pose to spawn the robot at"/>
    <arg name="initial_joint_positions" value="
        -J jLeftAnkle_rotx -0.3456
        -J jLeftAnkle_roty 0.541
        -J jLeftAnkle_rotz -0.2508
        -J jLeftBallFoot_roty -0.1745
        -J jLeftHip_rotx -0.5235
        -J jLeftHip_roty -1.8369
        -J jLeftHip_rotz 0.7853
        -J jLeftKnee_roty 1.3875
        -J jLeftKnee_rotz 0.0274
        -J jRightAnkle_rotx 0
        -J jRightAnkle_roty 0
        -J jRightAnkle_rotz 0
        -J jRightBallFoot_roty 0
        -J jRightHip_rotx 0.5235
        -J jRightHip_roty -1.8943
        -J jRightHip_rotz -0.7853
        -J jRightKnee_roty 0.6874
        -J jRightKnee_rotz 0

        -J jRightC7Shoulder_rotx 0.0872
        -J jRightElbow_roty 1.4835
        -J jRightElbow_rotz 1.0954
        -J jRightShoulder_rotx 0.2274
        -J jRightShoulder_roty -1.352
        -J jRightShoulder_rotz -0.082
        -J jRightWrist_rotx 0.3117
        -J jRightWrist_rotz 0.0321

        -J jL1T12_rotx 0.0271
        -J jL1T12_roty -0.0099
        -J jL4L3_rotx -0.1418
        -J jL4L3_roty 0.0099
        -J jL5S1_rotx 0.0611
        -J jL5S1_roty -0.0948"
    doc="Initial joint configuration of the robot"/>
    <include file="$(find moveit_human_config)/launch/gazebo.launch" pass_all_args="true" >      
    </include>
    <include file="$(find moveit_human_config)/launch/demo.launch" pass_all_args="true">
      <arg name="load_robot_description" value="false" />
      <arg name="moveit_controller_manager" value="ros_control" />
    </include>
    <node pkg="moveit_human_config" name="move_group_python_interface" type="human_controller.py" output="screen">
    </node>
  </group > -->

  <!-- <group ns="/fanuc_gazebo">
    <arg name="world_pose" default="-x -0.7 -y 0 -z 0 -R 0 -P 0 -Y 0" doc="Pose to spawn the robot at"/>
    <arg name="initial_joint_positions" value="
        -J joint_1 0.3"
        />
    <arg name = "robot_arm" default="robot_arm"/>
    <arg name="model" value="fanuc_gazebo"/>
    <include file="$(find moveit_fanuc_config)/launch/gazebo.launch" pass_all_args="true"/>
    <include file="$(find moveit_fanuc_config)/launch/demo.launch" pass_all_args="true">
      <arg name="load_robot_description" value="false" />
      <arg name="moveit_controller_manager" value="ros_control" />
    </include>
    <node pkg="moveit_fanuc_config" name="move_group_python_interface" type="fanuc_controller.py" output="screen">
    </node>
  </group>

  <group ns="/fanuc1_gazebo">
    <arg name="world_pose" default="-x +0.7 -y 0 -z 0 -R 0 -P 0 -Y 3.14159" doc="Pose to spawn the robot at"/>
    <arg name="initial_joint_positions" value="
    -J joint_1 0.3"
    />
    <arg name = "robot_arm" default="robot_arm"/>
    <arg name="model" value="fanuc1_gazebo"/>
    <include file="$(find moveit_fanuc1_config)/launch/gazebo.launch" pass_all_args="true"/>
    <include file="$(find moveit_fanuc1_config)/launch/demo.launch" pass_all_args="true">
      <arg name="load_robot_description" value="false" />
      <arg name="moveit_controller_manager" value="ros_control" />
    </include>
    <node pkg="moveit_fanuc1_config" name="move_group_python_interface" type="fanuc_controller.py" output="screen">
    </node>
  </group> -->
</launch>
